src/Tasks/Microsoft.NET.Build.Tasks/NuGetUtils.NuGet.cs

```diff
@@ -47,7 +47,167 @@ public static NuGetFramework ParseFrameworkName(string frameworkName)
             return frameworkName == null ? null : NuGetFramework.Parse(frameworkName);
         }
 
-        public static bool IsApplicableAnalyzer(string file, string projectLanguage)
+        /// <summary>
+        /// Gets a collection of 'analyzer' assets that should be excluded based on the
+        /// <paramref name="compilerApiVersion"/>.
+        /// </summary>
+        /// <remarks>
+        /// This allows packages to ship multiple analyzers that target different versions
+        /// of the compiler. For example, a package may include:
+        ///
+        /// "analyzers/dotnet/roslyn3.7/analyzer.dll"
+        /// "analyzers/dotnet/roslyn3.8/analyzer.dll"
+        /// "analyzers/dotnet/roslyn4.0/analyzer.dll"
+        ///
+        /// When the <paramref name="compilerApiVersion"/> is 'roslyn3.9', only the assets 
+        /// in the folder with the highest applicable compiler version are picked.
+        /// In this case,
+        /// 
+        /// "analyzers/dotnet/roslyn3.8/analyzer.dll"
```

Comment:
Is this comment correct? For 3.9 if should pick 4.0 right?

Reply:
I believe the comment is correct.  If we are running on Roslyn v3.9, we can't pick the `roslyn4.0` asset, or else we would get a load exception that `Microsoft.CodeAnalysis.dll version=4.0.0.0 is not found`.

Since `v4.0` is higher than the current version `v3.9`, we have to exclude it.

This works very similar to TFM selection: you can't pick a `net6.0` asset out of a NuGet package when the app is trying to run on `net5.0`.